python/pyarrow/tests/test_array.py

```diff
@@ -3409,6 +3409,17 @@ def test_array_accepts_pyarrow_array():
     assert arr == result
 
 
+def test_array_accepts_pyarrow_scalar():
```

Comment:
In `test_convert_builtin.py`, there is a `parametrize_with_collections_types` that can be used to test with multiple sequence types. 
(maybe move the test there to make it easier to use that, while scalars are not "built-in" python objects, it's still converting from a built-in sequence type of scalars ... )